lessons/231/python-app/main.py

```diff
@@ -86,43 +86,42 @@ async def create_device(
         now = datetime.datetime.now(datetime.timezone.utc)
         device_uuid = uuid.uuid4()
 
-        insert_query = """
-            INSERT INTO python_device (uuid, mac, firmware, created_at, updated_at)
-            VALUES ($1, $2, $3, $4, $5)
-            RETURNING id ;
+        async with conn.transaction():
```

Comment:
I don't think this is ok, for one: the GO implementation does not seem to operate in a transaction (I might be wrong, I weak at GO). Secondly this runs into the connection pool problem, you only need a connection for line 97, here you will return the connection on the exit (if that's what the transaction is for here) or on the exit from the Dependency. Also there's no commit.